spec/shared_behaviours/iterative_functions.rb

```diff
@@ -67,3 +59,25 @@
     end.to_not raise_error
   end
 end
+
+shared_examples_for 'all iterative functions with required block' do |func|
+
+  it 'raises an error when called without a block' do
+    expect do
+      compile_to_catalog(<<-MANIFEST)
+          [1].#{func}
+      MANIFEST
+    end.to raise_error(Puppet::Error, /expects a block/)
+  end
+end
+
+shared_examples_for 'all iterative functions that requires no block' do |func|
```

Comment:
In this context it means that they do not require a block. Whether or not they optionally accepts one isn't tested here.